ee/codegen/src/__test__/nodes/generic-nodes/node-display.test.ts

```diff
@@ -63,4 +64,76 @@ describe("NodeDisplay", () => {
 
     expect(await writer.toStringFormatted()).toMatchSnapshot();
   });
+
+  it("should skip Display class when icon and color match base class defaults", async () => {
+    const apiNodeDef = nodeDefinitions.nodes.find(
+      (node) => node.definition?.name === "APINode"
+    );
+    if (
+      !apiNodeDef?.definition?.module ||
+      !apiNodeDef.display_data?.icon ||
+      !apiNodeDef.display_data?.color
+    ) {
+      throw new Error("APINode definition not found in node-definitions.json");
+    }
+
+    const nodeData = genericNodeFactory({ label: "MyCustomAPINode" });
+    nodeData.base = {
+      name: "APINode",
+      module: apiNodeDef.definition.module,
+    };
+    nodeData.displayData = {
+      icon: apiNodeDef.display_data.icon,
+      color: apiNodeDef.display_data.color,
+    };
```

Comment:
ah we don't have generic node in our node-definitions.json. instead of editing the `nodeData`, let's just use the apiNodeFactory